### PR TITLE
OD-283 [Fix] "Mark maps" settings disappear when user changes data source in DS provider for Interactive Graphics

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-283
## Description
"Mark maps" settings dissaper when user chages data source
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cYljIwNHCy
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/interface/add-markers.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ Fliplet.InteractiveMap.component('add-markers', {
         })
       }
 
-      this.dataSourceId = ds.id
+      this.markersDataSource.id = ds.id
```
Comment (squallstar) [inline js/interface/add-markers.js:115]: please confirm this doesn't break if "markersDataSource" is undefined

Comment (AndrRyaz) [reply js/interface/add-markers.js]: @squallstar yes, your right i miss this point, i made changes and find another way

**Comment by squallstar** [comment]
@AndrRyaz please explain your last commit as I literally have no idea about why those changes were made and why they are required.

**Comment by AndrRyaz** [comment]
@squallstar  Yes sorry, in this line as shown here 
https://github.com/Fliplet/fliplet-widget-interactive-map/pull/96/files#diff-d3f7442d888676d4dac3d81003b5975ba2f8737c520b2152b174e959a8e9351cR108
I already added the changing current marker Data Source name because when the user clicks 'use these settings' the data source provider name stays old one.

At this line 
https://github.com/Fliplet/fliplet-widget-interactive-map/pull/96/files#diff-d3f7442d888676d4dac3d81003b5975ba2f8737c520b2152b174e959a8e9351cL285

I don't understand why this line is needed, because when the user changes the data source in the provider and clicks "use these settings", the data source id should change, but in this line it changes back to old id

**Comment by squallstar** [comment]
@tonytlwu seeing the above, do you (or @hcarneiro, which created the component) see what's going on?

**Comment by tonytlwu** [comment]
It isn't clear to me why we should or could remove `this.markerDataSourceId = ds.id`. The line has been there for 2 years and was placed by @hcarneiro in https://github.com/Fliplet/fliplet-widget-interactive-map/commit/40d03d9e0a10c8c5b957b7128ea194aad7806be6 @hcarneiro ?

**Comment by hcarneiro** [comment]
@tonytlwu @squallstar I don't know why we would like that removed.

I believe that `this.markersDataSource` is updated in this line when the user closes the data sources overlay: https://github.com/Fliplet/fliplet-widget-interactive-map/pull/96/files#diff-d3f7442d888676d4dac3d81003b5975ba2f8737c520b2152b174e959a8e9351cL790

Therefore if that that line is removed when we use the new settings, the `this.dataSourceId` is never updated to a new ID.

Unless I'm not getting something, that's what was supposed to do.

**Comment by AndrRyaz** [comment]
@hcarneiro 
I'm a little misled because in the `useSettings` function `this.markersDataSource.id` has an old data source id but in `markersDataSourceChange` function : https://github.com/Fliplet/fliplet-widget-interactive-map/pull/96/files#diff-d3f7442d888676d4dac3d81003b5975ba2f8737c520b2152b174e959a8e9351cR117
`this.dataSource.id` is getting the new id and it doesnt matter that this is new data source or a data source that has already been

**Comment by hcarneiro** [comment]
@AndrRyaz so you are saying that when the overlay closes this doesn't assign the new data source to `this.markerDataSource` ?
Here: https://github.com/Fliplet/fliplet-widget-interactive-map/blob/master/js/interface/add-markers.js#L790

**Comment by AndrRyaz** [comment]
@hcarneiro i'm talking about this part of scenario
https://screencast-o-matic.com/watch/cYl66YNRqP

**Comment by hcarneiro** [comment]
@AndrRyaz Just tested as well and can confirm that the ID does indeed revert back to the old one. Not sure why, not sure if it was because the new provider was implemented, but you are right and therefore that line should be removed.

Please also make sure that after changing the data source and the users clicks on "View data source", when they close the data source overlay, the data source ID is still correct when the click on "Use these settings".

@tonytlwu also this is a very weird UI:
image

**Comment by AndrRyaz** [comment]
@hcarneiro, I tested all scenarios and can confirm everything is working fine
https://screencast-o-matic.com/watch/cYl6X1NROW

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Medium
Affected: Studio users configuring interactive graphic components
Verify: Data source changes are correctly saved and loaded

Squash and merge:

**Pull request merged**